apps/amo/urlresolvers.py

```diff
@@ -163,6 +163,10 @@ def fix(self, path):
         path = path.lstrip('/')
         url_parts = [self.request.META['SCRIPT_NAME']]
 
+        if path.startswith(settings.SUPPORTED_NONAPPS_NONLOCALES_DEEP):
```

Comment:
See comment below: if the settings is instead an iterable, this won't work.

It seems we're already doing something along those lines (not redirecting) if you look at the following test: https://github.com/mozilla/olympia/pull/877/files#diff-ddb145258faf9e332785b9f2d3660949R79
Would it make sense to do something similar, or reuse this code? (disclaimer, I haven't read the code nor checked what it's doing exactly).